### PR TITLE
Enable get_image_urls without 'rows' input

### PR DESCRIPTION
**Pull request by sofiacbeco**

### CODE REVIEW

File: zegami_sdk/collection.py [hunk 1]
```diff
@@ -260,14 +260,17 @@ def get_image_urls(self, rows, source=0, generate_signed_urls=False,
...
+        # If 'rows' are not defined, get all rows of collection.
         if type(rows) == pd.DataFrame:
             indices = list(rows.index)
 
         elif type(rows) == list:
             indices = [int(r) for r in rows]
         elif type(rows) == int:
             indices = [rows]
+        elif not rows: 
+            indices = [i for i in range(len(self))]
```
Comment (ahaith) [inline zegami_sdk/collection.py:273]: Actually, it's not `indices` that needs to be set in this case, but `imageset_indices` (line 279)
The images are in a completely different order than the rows, which is there is a lookup, but in this case you would bypass that lookup entirely and just build a list of all indices.

Comment (ahaith) [reply zegami_sdk/collection.py:273]: In fact, you can ignore this comment - there is a lookup when there's no data, which just does a trivial n->n mapping.

File: zegami_sdk/collection.py [hunk 2]
```diff
@@ -260,14 +260,17 @@ def get_image_urls(self, rows, source=0, generate_signed_urls=False,
         By default the uploaded images are fetched, but it's possible to fech e.g. the thumbnails
         only, by providing an alternative imageset id.
         """
-        # Turn the provided 'rows' into a list of ints
+        # Turn the provided 'rows' into a list of ints.
+        # If 'rows' are not defined, get all rows of collection.
         if type(rows) == pd.DataFrame:
```
Comment (fayaaz) [inline zegami_sdk/collection.py:265]: Small suggestion for restructuring logic:

```suggestion
        if not rows:
             < empty row logic >
        else:
            if type(rows) == pd.DataFrame:
                ...
            elif ...
            else:
                raise ValueError('Invalid rows argument, \'{}\' not supported'.format(type(rows)))            
```

File: zegami_sdk/collection.py [hunk 3]
```diff
@@ -260,14 +260,17 @@ def get_image_urls(self, rows, source=0, generate_signed_urls=False,
...
             indices = list(rows.index)
 
         elif type(rows) == list:
             indices = [int(r) for r in rows]
         elif type(rows) == int:
             indices = [rows]
+        elif not rows: 
+            indices = [i for i in range(len(self))]
         else:
             raise ValueError('Invalid rows argument, \'{}\' not supported'.format(type(rows)))
```
Comment (fayaaz) [inline zegami_sdk/collection.py:275]:
```suggestion
        if rows:
            if type(rows) == pd.DataFrame:
                indices = list(rows.index)
            elif type(rows) == list:
                indices = [int(r) for r in rows]
            elif type(rows) == int:
                indices = [rows]   
            else:
                raise ValueError('Invalid rows argument, \'{}\' not supported'.format(type(rows)))
        else: 
            indices = [i for i in range(len(self))]
```

**Review by fayaaz — COMMENTED** [review]
Slight restructure to logic

File: zegami_sdk/collection.py (hunk 1 above)
Comment (ahaith) [reply zegami_sdk/collection.py]: As discussed, that doesn't help if it's a collection which *does* have rows, but `None` is passed in. Behaviour in that case possibly pretty erroneous. Could address that with a follow up.

Comment (Vaspra) [reply zegami_sdk/collection.py]: I think this is better explicit. If you want all rows, you simply need to do `c.get_image_urls(rows=c.rows)`

Comment (sofiacbeco) [reply zegami_sdk/collection.py]: This change was to be able to get urls from a collection with no rows (no metadata). An alternative could be to make the default rows valid/readable through the sdk.

Comment (ahaith) [reply zegami_sdk/collection.py]: Accessing the *output* rows would make a lot of sense. A simple way to achieve this only for image-only collections would be to tweak the `_dataset_id` getter to return `self._data['output_dataset_id']`, but I don't think we'd want to do this across the board as we'd risk people baking e.g. the image similarity columns into the uploaded data.

A few options:
1. Make an explicit distinction between `input data` and `output data` within the sdk. Both are available and users should grab whichever is appropriate to their needs.
2. If the input data is non-existent, fall back on the output data
3. Always use the output data

Would be good to hear views on this from both @Vaspra  and @sofiacbeco

**Pull request merged**